lib/Dialect/Transforms/DropSymbolicShapeOpsPass.cpp

```diff
@@ -0,0 +1,61 @@
+//===------------------------------------------------------------*- C++ -*-===//
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// Also available under a BSD-style license. See LICENSE.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir-tcp/Dialect/Transforms/DropSymbolicShapeOpsPass.h"
+
+#include "mlir-tcp/Dialect/IR/TcpDialect.h"
+#include "mlir-tcp/Dialect/IR/TcpOps.h"
+#include "mlir-tcp/Dialect/Transforms/Passes.h"
+
+#include "./PassDetail.h"
+
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+using namespace mlir;
+
+namespace mlir::tcp {
+
+namespace {
+
+class RemoveBindSymbolicShapeOps
+    : public OpRewritePattern<tcp::BindSymbolicShapeOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(tcp::BindSymbolicShapeOp op,
+                                PatternRewriter &rewriter) const override {
+    if (!op->getUsers().empty())
```

Comment:
Is this ever true?
Isn't users going to be always empty as this op is used like `tcp.bind_symbolic_shape %arg0, [%0], affine_map<()[s0] -> (s0)>` where the returned value is ignored

Reply:
Yes this should never be true; I wanted an assertion here instead of a match failure, but I'll just remove this since the default dialect verifier should suffice.